io-classes/src/Control/Concurrent/Class/MonadMVar.hs

```diff
@@ -181,8 +187,22 @@ instance ( MonadMask m
     modifyMVarMasked (WrappedMVar v) f = ReaderT $ \r ->
       modifyMVarMasked v (\a -> runReaderT (f a) r)
 
+--
+-- MonadInspectMVar
+--
 
-
+-- | This type class is intended for
+-- ['io-sim'](https://hackage.haskell.org/package/io-sim), where one might want
```

Comment:
oh, does haddock accept markdown style links?

Reply:
It does :)

Reply:
Yeah, I copied this from `MonadInspectSTM`!